corpus/accounts/models.py

```diff
@@ -151,3 +151,84 @@ def save(self, *args, **kwargs):
 
     def __str__(self):
         return f"{self.user.first_name} {self.user.last_name} [{self.sig.name}]"
+
+
+class Core(models.Model):
+
+    POST_CHOICES = (
+        (1, "Convenor"),
+        (2, "Chairperson"),
+        (3, "Vice Chairperson"),
+        (4, "Secretary"),
+        (5, "Joint Secretary"),
+        (6, "Treasurer(Branch)"),
+        (7, "Treasurer(Institute)"),
+        (8, "Webmaster"),
+        (9, "Media Lead"),
+        (10, "Outreach Lead"),
+        (11, "Envision Lead"),
+        (12, "Labs Lead"),
+        (13, "CompSoc Chair"),
+        (14, "CompSoc Vice Chair"),
+        (15, "CompSoc Secretary"),
+        (16, "CompSoc Project Head"),
+        (17, "CompSoc Project Coordinator"),
+        (18, "CIS Chair"),
+        (19, "CIS Secretary"),
+        (20, "CIS Project Head"),
+        (21, "Diode Chair"),
+        (22, "SPS Chair"),
+        (23, "SPS Vice Chair"),
+        (24, "SPS Secretary"),
+        (25, "CAS Chair"),
+        (26, "CAS Vice Chair"),
+        (27, "CAS Secretary"),
+        (28, "RAS Chair"),
+        (29, "RAS Secretary"),
+        (30, "Piston Chair"),
+        (31, "Piston Vice Chair"),
+        (32, "Piston Secretary"),
+        (33, "IAS Chair"),
+        (34, "IAS Secretary"),
+    )
+    executivemember = models.OneToOneField(
+        ExecutiveMember, null=False, on_delete=models.CASCADE
+    )
+    post = models.IntegerField(null=False, choices=POST_CHOICES)
+    sig = models.ForeignKey(SIG, null=False, on_delete=models.CASCADE)
+    term_start = models.DateField()
+    term_end = models.DateField()
+
+    def __str__(self):
+        self_user = self.executivemember.user
+        return f"{self_user.first_name} {self_user.last_name}"
+
+    def get_post_display(self):
+        return dict(Core.POST_CHOICES).get(self.post)
+
+
+class Faculty(models.Model):
+    class Meta:
+        verbose_name_plural = "faculties"
+
+    FACULTY_POSTS = (
```

Comment:
Is there a reason why Faculty post choices are 1-1 mapped with their text but Core post choices are integer mapped?

Reply:
I'm sorting the Core posts but not the faculty posts. So every core post corresponds to an integer key and thus I sort them using the keys. But the same is not really needed for Faculty posts so I let it be. Should I do the same for Faculty posts?